src/main/java/org/jenkins_ci/plugins/flexible_publish/ConditionalPublisher.java

```diff
@@ -169,20 +170,41 @@ public ConditionalPublisherDescriptor getDescriptor() {
     }
 
     public boolean prebuild(final AbstractBuild<?, ?> build, final BuildListener listener) {
-        for (BuildStep publisher: getPublisherList()) {
-            if (!runner.prebuild(condition, publisher, build, listener)) {
-                return false;
-            }
+        if (getPublisherList().size() <= 0) {
+            return true;
+        } else if (getPublisherList().size() > 1) {
+            return runner.prebuild(condition, new RunAllBuilder(getPublisherList()), build, listener);
         }
-        return true;
+        return runner.prebuild(condition, getPublisherList().get(0), build, listener);
     }
 
     public boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher, final BuildListener listener)
                                                                                                 throws InterruptedException, IOException {
-        for (BuildStep publisher: getPublisherList()) {
-            if (!runner.perform(condition, publisher, build, launcher, listener)) {
+        if (getPublisherList().size() <= 0) {
+            return true;
+        } else if (getPublisherList().size() > 1) {
+            return runner.perform(condition, new RunAllBuilder(getPublisherList()), build, launcher, listener);
```

Comment:
IMO the change is valid in general. The initial behavior conflicts with the core's one.
BTW, @KostyaSha is right that the behavior may break something. What about introducing "execution strategy" as in Advanced Configuration? In such case it could be possible to leave the original behavior as is,